### DISCUSSION TITLE
How to use Quick Terminal?

### DISCUSSION THREAD

**Discussion by trevor-e** [ghostty-org/ghostty#3610] [Q&A] [answered]
I see Quick Terminal is listed as a feature on the website, the top feature for macOS. And I see it as an option in the menu at View > Quick Terminal. This seems like a popular feature so I must be missing something very obvious -- how do I use it? 

There's no hotkey listed next to it in the View dropdown, and I can't find any other way to trigger it. Having to select View > Quick Terminal seems very... not quick lol. Why would I do this over opening a new tab? Would someone please mind explaining how to use this feature, thanks! 😅

**Comment by edrozenberg** [comment, marked as answer]
You can add a binding for this in the `config` file, something like this for ex:
`keybind = global:ctrl+grave_accent=toggle_quick_terminal`

(this one binds CTRL + backtick to the Quick Terminal).

The `global` item means it will work even if some other app is active, but (on MacOS) `global` will require Ghostty to have Privacy->Accessibility permissions in the system settings in order to work. Otherwise not using `global` means Ghostty needs to be active in order for the keybinding to work.

Tap the binding once to show, tap again to hide. Or, with the default config, clicking/focusing some other window will also auto-close the Quick Terminal (its contents will persist). You can also drag the edge of the Quick Terminal to make it taller.

**Reply by trevor-e** [reply]
Thanks! This is very neat.

**Reply by mityukov** [reply]
It looses the hotkey from time to time (several times a day, actually)… I need to launch Ghostty from  the dock and close its regular window -- after that the quick terminal starts responding to the hotkey again. :(

**Comment by daletman** [comment]
Since Ghostty has a Zero Configuration Philosophy I would argue there should be a default key binding since without a key binding this feature doesn't make much sense?!

https://ghostty.org/docs/config#zero-configuration-philosophy

**Reply by edrozenberg** [reply]
I agree but can't think of a shortcut that would work well for most people, esp. one that you can make global without stepping on shortcuts used by other apps. On Mac I settled on `Ctrl Opt Cmd + Space`, nothing else is using that atm for me. I don't suggest using `Cmd-T` or `Cmd-Q` on Mac :).

Previously I looked into using a sided modifier for ex. the right side CTRL key:
`keybind = ctrl+right_control=toggle_quick_terminal`

but this doesn't function properly atm, possibly due to a Ghostty bug
(https://github.com/ghostty-org/ghostty/discussions/3734)

For users who have F11/F12 available, something like this would work:
`keybind = f11=toggle_quick_terminal`

(in my case I'm using both of these Fn's for other things on Mac)

**Reply by trevor-e** [reply]
After learning more about the feature and reading the docs for it, IMO just making the https://ghostty.org/docs/config/keybind/reference#toggle_quick_terminal docs more prominent would be very helpful and simple, like having a link to this from somewhere on the main website. It gets lost with the rest of the documentation and requires some user setup to work.

**Reply by sebastian-correa** [reply]
I use `Cmd + Esc` for the same thing in iTerm. I have no clue whether I gave up some MacOS thing, but evidently I don't care enough for it to be annoying in my use case xD.

**Reply by oranges13** [reply]
I use ctrl + ` for iterm also.

**Reply by pluiedev** [reply]
Config defaults should be discussed at #2363

**Reply by dlamblin** [reply]
> I use `Cmd + Esc` for the same thing in iTerm. I have no clue whether I gave up some MacOS thing, but evidently I don't care enough for it to be annoying in my use case xD.

FYI ⌘-Esc and ^\` aren't assigned to system-wide shortcuts. Though ⌘-\` is the default for system-wide "move focus to next window". Similar to ⌘-Tab cycling though applications, the default would cycle through windows within that application. It's pretty handy. VS Code uses ^\` for the internal terminal, which Zed also uses. While Zed uses Esc, ^Esc, ⌘-Esc to close menus.

**Comment by aaronik** [comment]
Is there a way to open ghostty in quick terminal mode by specifying a flag? Like ghostty --quick-terminal? I'd rather manage the keybindings in my operating system than have ghostty manage it from the config file. That way I don't have to grant any extra permissions, and I don't have to fumble with finding the correct key combo strings.

**Reply by natikgadzhi** [reply]
I didn't figure it out yet, and settled on having a global keybind (outside of Ghostty) to open Ghostty itself for now.

I _think_ this would be a welcome addition (I am not working on Ghostty and can't speak for Mitchell though), and you can probably craft something like a new CLI action (akin to what +list-themes etc do) as a launcher for the behavior. If we had `ghostty +quick-terminal`, it would be trivial to trigger it with your keybind tool of choice.

**Reply by JakeWharton** [reply]
A CLI flag would be covered by #2353

**Reply by PandorasFox** [reply]
A flag to invoke this is necessary in my case: the Wayland compositor I use (Niri) has wlr_shell support, but no global shortcuts support. Adding a bind for this in Niri is very doable (and honestly preferable)

**Reply by pluiedev** [reply]
Again, this is #2353 territory.

**Comment by elliotdes** [comment]
Is there a way for this to work while Ghostty is terminated? Currently for quick terminal to work, Ghostty must be active (appear in the app switcher). I do not always have my terminal app open, so being able to use quick terminal any time would be useful.

**Reply by mattmorgis** [reply]
On macOS, go to Privacy & Security > Accessibility then add Ghostty.app

**Reply by tomcarman** [reply]
I've tried adding Ghostty.app under Privacy & Security > Accessibility, but seems to still require Ghostty to be open in the Dock for it to work (although it works if Ghostty is not the active window).

**Reply by edrozenberg** [reply]
> Is there a way for this to work while Ghostty is terminated?

Not currently but someone mentioned this may come in the future. On Mac I'm assuming this would be as a background service or daemon, would be great.

**Comment by smikheiev** [comment]
is there a way to change the default quick terminal size (width/height)?

**Reply by edrozenberg** [reply]
Not yet but there's an open issue for it: https://github.com/ghostty-org/ghostty/issues/2384

Currently (see https://ghostty.org/docs/config/reference) you can specify:
`quick-terminal-position`
> The position of the "quick" terminal window
Valid values are:
top - Terminal appears at the top of the screen.
bottom - Terminal appears at the bottom of the screen.
left - Terminal appears at the left of the screen.
right - Terminal appears at the right of the screen.
center - Terminal appears at the center of the screen.

If you set position to `center` you will get a quick term window that is centered and also takes up less than full width, you may prefer this.

You can also manually adjust the height of a for ex. `top` quick term by dragging its bottom edge (not persistent across app relaunches).

**Reply by smikheiev** [reply]
thanks!

**Reply by ilyagr** [reply]
When the position is "center", I haven't found any way to change the terminal size at all. In particular, dragging and edge/corner does not seem to work for me.

The `window-height` option does not seem to have any effect either.

OTOH, #2384 seems to be actively worked on, which is great! :)

**Reply by 00-kat** [reply]
Open a [new Discussion] please!

[new Discussion]: https://github.com/ghostty-org/ghostty/discussions/new?category=issue-triage

**Comment by mrkbac** [comment]
Is there a way to convert the quick terminal, to a normal window?

**Reply by edrozenberg** [reply]
> Is there a way to convert the quick terminal, to a normal window?

No, it would be nice :). Note that quick terminal already has most of the features of a normal window (other than titlebar, tabs, dragging, or fullscreen). For ex. you can create splits in it.

Regarding tabs and other limitations of the quick terminal:
https://github.com/ghostty-org/ghostty/pull/2320
> No tabs in the quick terminal (fixable with a custom tab bar)

**Comment by amk221** [comment]
@mrkbac yeh tabs would be great and help me get off iterm. Currently if you do make a new tab the UI gets a bit confused

**Reply by edrozenberg** [reply]
There are no tabs in quick terminal, what happens w/ Cmd-T while in quick terminal is it creates a new window the same size as the quick terminal :), it's a quirk/bug atm, it shouldn't do anything.

**Comment by pluiedev** [comment]
Closing as resolved since the original question has already been answered. Please ask new questions relating to this topic in new discussions.
